include/uv.h

```diff
@@ -45,7 +45,11 @@ extern "C" {
 #   define UV_EXTERN /* nothing */
 # endif
 #elif __GNUC__ >= 4
+# if defined(BUILDING_UV_SHARED)
 # define UV_EXTERN __attribute__((visibility("default")))
+# else
+# define UV_EXTERN /* nothing */
+# endif
```

Comment:
I think this breaks cmake builds where an executable links statically to libuv but re-exports its API (ex: node, except node doesn't use cmake.)

What is the motivation for opening this PR? Did something not work?

Reply:
@nmgwddj please elaborate instead of marking resolved without comment